resources/fetchersManager/data.go

```diff
@@ -57,19 +57,20 @@ func NewData(log *logp.Logger, interval time.Duration, timeout time.Duration, fe
 
 // Run starts all configured fetchers to collect resources.
 func (d *Data) Run(ctx context.Context) Stop {
-	go d.fetchAndSleep(ctx)
+	fetchContext, cancel := context.WithCancel(ctx)
```

Comment:
lifted the context to be shared with `d.stopData` so we can select a case to return from: `After(grace)` / `ctx.Done()`